browser/src/layer/tile/CommentListSection.ts

```diff
@@ -1677,14 +1687,18 @@ export class CommentSection extends CanvasSectionObject {
 	// Returns the sub-root comment index of given id
 	private getSubRootIndexOf (id: any): number {
 		var index = this.getIndexOf(id);
-		var comment = this.sectionProperties.commentList[index];
-		var parentId = comment.sectionProperties.data.parent;
 
-		while (index >= 0) {
-			if (this.sectionProperties.commentList[index].sectionProperties.data.id !== parentId && this.sectionProperties.commentList[index].sectionProperties.data.parent !== '0')
-				index--;
-			else
-				break;
+		if (index !== -1)
+		{
+			var comment = this.sectionProperties.commentList[index];
+			var parentId = comment.sectionProperties.data.parent;
+
+			while (index >= 0) {
+				if (this.sectionProperties.commentList[index].sectionProperties.data.id !== parentId && this.sectionProperties.commentList[index].sectionProperties.data.parent !== '0')
```

Comment:
Can we have:
`this.sectionProperties.commentList[index].sectionProperties.data.parent` -> `this.sectionProperties.commentList[index].getParentId()`

`this.sectionProperties.commentList[index].sectionProperties.data.id` -> `this.sectionProperties.commentList[index].getId()`

and maybe helper methods:
`getCommentByIndex(index) { return this.sectionProperties.commentList[index]; }`
`getCommentById(id) { return ... }`

I assume `comment index` != `comment id`

This will improve code readability a lot. Let's introduce usage of it only in this method. But will be good to use it later when doing any rework.